EDU-18419 - Creating an extension point (#2678)
vtexdocs/dev-portal-content: docs/guides/VTEX-Sales-App/creating-a-vtex-sales-app-extension.md

---
title: "Creating a VTEX Sales App extension"
slug: "creating-a-vtex-sales-app-extension"
hidden: false
excerpt: "Learn how to create a VTEX Sales App extension in your FastStore project and render it in a predefined extension point."
createdAt: "2026-05-25T16:10:21.214Z"
---

After setting up a VTEX Sales App extensions project in your [FastStore monorepo](https://developers.vtex.com/docs/guides/faststore/monorepo-overview), you're ready to create custom experiences by rendering React components in predefined extension points.

> ℹ️ To create a VTEX Sales App extension, it's not mandatory to configure the `discovery` module. In this setup, FastStore provides the monorepo structure, tooling, and build process, while the `sales-app` module contains the extension code.

In this guide, you'll learn how to create a basic extension, connect it to an extension point, style it with CSS, and understand how the deployment flow works.

## Before you begin

> * Complete the [Setting up extensions for VTEX Sales App](https://developers.vtex.com/docs/guides/setting-up-extensions-for-vtex-sales-app) guide.

Author: https://github.com/vtexdocs/dev-portal-content/pull/2672

> * Make sure you have installed the FastStore WebOps. For detailed instructions, see the guide [Installing FastStore WebOps](https://developers.vtex.com/docs/guides/installing-webops-for-sales-app-extensions).

Author: https://github.com/vtexdocs/dev-portal-content/pull/2680

## Instructions

Suppose you want to create a custom message for your VTEX Sales App. Based on the sample folder generated in your monorepo when creating the project, follow the steps.

### Step 1 - Creating the component

To create a new component, add the following content to your TypeScript file (for example, `src/components/CustomMessage.tsx`):

> ```tsx

Author: ```suggestion
```typescript src/components/CustomMessage.tsx
```

export const CustomMessage = () => {
  return (
      <p>
        Extension example
      </p>
  )
}
```

This component renders a simple custom message in the VTEX Sales App interface.

### Step 2 - Registering the extension

Import the new component into `src/index.tsx`. Then, connect it to an extension point.

> ```tsx

Author: ```suggestion
```typescript src/index.tsx
```

import { defineExtensions } from '@vtex/sales-app';
import { Example } from './components/Example';
import { CustomMessage } from './components/CustomMessage';

export default defineExtensions({
  'cart.cart-list.after': CustomMessage,
});
```

> In this example, the component is rendered in the `cart.cart-list.after` extension point. Learn more about the available extension points in the [VTEX Sales App extension points](#LINK) guide.

Author: ```suggestion
In this example, the component is rendered in the `cart.cart-list.after` extension point. Learn more about the available extension points in the [VTEX Sales App extension points](https://developers.vtex.com/docs/guides/vtex-sales-app-extension-points) guide.
```

After saving the file, open the cart page again. The custom message should appear below the cart item list.

![VTEX cart page with empty cart and Extension example area displayed without custom styling](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@EDU-18419-creating-extension-point/images/creating-extensions-example-without-css.webp)

### Step 3 - (Optional) Styling the extension

To style your component, create a CSS file in the same folder as the component. For example:

```css
.custom-message {
  background: #000;
  color: #fafafa;
  padding: 16px;
}
```

Then update `CustomMessage.tsx` to import the stylesheet and apply the class.

> ```tsx

Author: ```suggestion
```typescript src/components/CustomMessage.tsx
```

---
title: "Creating a VTEX Sales App extension"

> slug: creating-a-vtex-sales-app-extension

Author: ```suggestion
slug: "creating-a-vtex-sales-app-extension"
```

hidden: false
excerpt: "Learn how to create a Sales App extension and render it in a predefined extension point."
createdAt: "2026-05-28T00:00:00.000Z"
---

After setting up a Sales App extension project in your [FastStore monorepo](https://developers.vtex.com/docs/guides/faststore/monorepo-overview), you're ready to create custom experiences by rendering React components in predefined extension points.

> ℹ️ To create a Sales App extension, it's not mandatory to configure the `discovery` module. In this setup, FastStore provides the monorepo structure, tooling, and build process, while the `sales-app` module contains the extension code.

In this guide, you'll learn how to create a basic extension, connect it to an extension point, and style it with CSS.

## Before you begin

> * Complete the [Setting up extensions for VTEX Sales App](https://developers.vtex.com/docs/guides/setting-up-sales-app-extensions) guide.

Author: ```suggestion
* Complete the [Setting up Sales App extensions](https://developers.vtex.com/docs/guides/setting-up-sales-app-extensions) guide.
```

---
title: "Creating a VTEX Sales App extension"
slug: creating-a-vtex-sales-app-extension
hidden: false
excerpt: "Learn how to create a VTEX Sales App extension in your FastStore project and render it in a predefined extension point."
createdAt: "2026-05-22T16:10:21.214Z"
---

After setting up a VTEX Sales App extensions project in your FastStore monorepo, you're ready to create custom experiences by rendering React components in predefined extension points.

In this guide, you'll learn how to review the generated project structure, create a basic extension, connect it to an extension point, and style it with CSS.

## Before you begin

> Complete the [Setting up extensions for VTEX Sales App in FastStore web stores](https://developers.vtex.com/docs/guides/setting-up-extensions-for-vtex-sales-app-in-faststore-web-stores) guide.

Author: https://github.com/vtexdocs/dev-portal-content/pull/2672

## Instructions

Consider you want to create a custom counter for your VTEX Sales App. Based on the sample folder generated in your monorepo when creating the project, follow the steps below.

> ℹ️ Replace example values such as `store-a` and folder paths with the values from your own project.

### Step 1 - Creating the component

To create a new component, add the following content to your TypeScript file (for example, `src/components/CustomMessage.tsx`):

```tsx
export const CustomMessage = () => {
  return (
      <p>
        Extension example
      </p>
  )
}
```

This component renders a simple custom message in the VTEX Sales App interface.

### Step 2 - Register the extension

Import the new component into `src/index.tsx`. Then, connect it to an extension point.

```tsx
import { defineExtensions } from '@vtex/sales-app';
import { Example } from './components/Example';
import { CustomMessage } from './components/CustomMessage';

export default defineExtensions({
  'cart.cart-list.after': CustomMessage,
});
```

> In this example, the component is rendered in the `cart.cart-list.after` extension point. Learn more about the available extension points in the [VTEX Sales App extension points](#LINK) guide.

Author: https://github.com/vtexdocs/dev-portal-content/pull/2673

After saving the file, open the cart page again. The custom message should appear below the cart item list.

![VTEX cart page with empty cart and Extension example area displayed without custom styling](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@EDU-18419-creating-extension-point/images/creating-extensions-example-without-css.webp)

### Step 3 - (Optional) Style the extension

To style your component, create a CSS file in the same folder as the component. For example:

```css
.custom-message {
  background: #000;
  color: #fafafa;
  padding: 16px;
}
```

Then update `CustomMessage.tsx` to import the stylesheet and apply the class.

```tsx
import { useSettings } from "@vtex/sales-app"
import './custom-message.css'

export const CustomMessage = () => {
  const { branding } = useSettings()
  return (
    <p className='custom-message'>
      Extension example
    </p>
  )
}
```

After saving the file, return to the browser to preview the styled extension.

![VTEX cart page with empty cart and Extension example area styled as a black bar at the bottom

> ](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@EDU-18419-creating-extension-point/images/creating-extensions-example-with-css.webp)

Author: ```suggestion
![VTEX cart page with empty cart and Extension example area styled as a black bar at the bottom](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@EDU-18419-creating-extension-point/images/creating-extensions-example-with-css.webp)
```
